### ISSUE TITLE
Add a button for requesting help or reporting error

### ISSUE THREAD

**Issue by halatmit** [mit-cml/appinventor-sources#2451]
**Describe the desired feature**

Add a button to the App Inventor screen that lets users type in a request for help.  It should probably just forward the request to the forum.  But first, it should collect some information, notably whether this is about Android or iOS (or both).

This could help minimize confusion in responding to support requests.

**Comment by nirvasangani** [comment]
@halatmit 

Sir there is already this two action is present in current app inventor screen under **Help** section but currently these action is not collecting any data from user and just forwarding user into community.

As per my understanding,

We can create a Pop-Up which show on **Report an Issue** item clicked and collect required data whether it is about android or iOS and after that forward user to community.

To achieve this functionality we need to use GWT and discourse API.

Thank you

Preet Vadaliya

**Comment by halatmit** [comment]
Good suggestion.
Thanks.

== Hal

Hal Abelson
***@***.***
Prof. of Comp. Sci. and Eng.
MIT Dept. of Elec. Eng. and Comp. Sci.

**Comment by PeterMathijssen** [comment]
I would hope that before users would enter a request for help that they would check the community to see if their question/problem was asked before. We see already a lot of users asking the same kind of questions on the community before even searching for it.

**Comment by ewpatton** [comment]
I agree with @PeterMathijssen here. Most of the problems encountered have likely already been solved elsewhere on the forum. Discourse has a mechanism for identifying "similar posts", so if based on what the user types we could direct them to a relevant post first, this would save power users and staff time repeating answers to similar inquiries.

**Comment by shreelakshmijoshi** [comment]
Hi @halatmit !
I am a new open source contributor and would like to solve this issue, could you help me understand the issue better by providing some extra references ?
Is this is issue open to contribute ?

**Comment by Shyam-123pandey** [comment]
Hi team,

This feature sounds like a useful way to reduce friction for users who need help but aren’t sure how or where to ask. Many beginners, especially students and teachers, may not know what information is important when reporting an issue, which can lead to confusion and slower responses on the forum.

Providing a dedicated “Request Help / Report Issue” button directly in the App Inventor UI could guide users through a short, beginner-friendly flow before redirecting them to the forum. Collecting basic context such as platform (Android, iOS, or both), project type, and possibly App Inventor version upfront would help ensure support requests are clearer and easier to respond to.

This approach keeps the forum as the central support channel while improving the quality of incoming questions and reducing back-and-forth for clarification. I’d be interested in exploring where such a button could fit naturally in the UI and how much information should be collected to remain helpful without overwhelming users.

Happy to investigate further if this aligns with the project direction.

**Comment by Shyam-123pandey** [comment]
Please reply so that i can know whether i am going in right direction or need to change approach thanks

**Comment by Shyam-123pandey** [comment]
Hello Sir just tell whether i am going in right direction or not
